### DISCUSSION TITLE
S32K14x Eval board with NHD 7 inch Display module (FTD812) Sample Code

### DISCUSSION THREAD

**Discussion by suryazoox** [RudolphRiedel/FT800-FT813#78] [General]
Hello @RudolphRiedel,
Thank you for helping me in advance. 
 
I am trying to use NHD 7 inch display module ( with FTDI 812) and S32K144 eval board to bring up the display module and to play around with it. do you have any sample code or projects that I could try using s32 Design studio to start playing around with it? 
Also, I tried using the following 
[EVE_Test_S32K144.zip](https://github.com/RudolphRiedel/FT800-FT813/files/10816338/EVE_Test_S32K144.zip) but I get the following errors: 
![image](https://user-images.githubusercontent.com/111003941/220981994-2c67a65d-b2a8-45d2-b5c4-5974c811c959.png)
I am currently using s32 design studio version 2.2. 

It would be great if you could also post some picture about your set up as well. 

Thank you very much in advance.

**Comment by RudolphRiedel** [comment]
Ugh, I was kind of hoping to be done with S32K. :-)
Well, for a couple of different reasons.

I still have the "S32 Design Studio for S32 Platform" installed and it says it is Version 3.4.
Well, it also tells me that there are updates available.
But the project that got opened automatically did still compile and the target is S32K144, I also still have the eval board here.

I really has been a while, I have not updated the library in the project to the current release so far.

**Reply by suryazoox** [reply]
Thanks for your response. But do not get done with S32K. I would really need your help with it. 

Well, I created a new project and started to copy the following header and c files. 
![image](https://user-images.githubusercontent.com/111003941/221012241-7060162f-628f-435d-a423-ed1365f3970c.png)
It reduced the number of errors but I am stuck with the following errors as shown below: 
![image](https://user-images.githubusercontent.com/111003941/221012372-e5c49d9a-4450-4607-adcf-304774c446f5.png)

Any help on this, please? 

Also, do we need to include the PD pin as well for the connection?

**Comment by RudolphRiedel** [comment]
I wrote that I was hoping to be done, not that I am done with S32K. :-)
Ok, I am updating my S32DS to 3.4.1 now and then I'll try to update the project to 5.0.6 of my library.

The issue with DL_CLEAR_RGB is that it was renamed to DL_CLEAR_COLOR_RGB in the more recent EVE.h.
g_pin_mux_InitConfigArr0 is declared in pin_mux.c from NXP.
NUM_OF_CONFIGURED_PINS0 is defined in pin_mux.h from NXP, somehow your version is missing "0" at the end?
I removed the function macro RESTORE_CONTEXT and added the define DL_RESTORE_CONTEXT instead.
 EVE_cmd_dl_burst(RESTORE_CONTEXT()); -> EVE_cmd_dl_burst(DL_RESTORE_CONTEXT);

And yes, I highly recommend using PD to the connections in order to get a proper reset cycle for init.
If PD is not used a EVE_cmdWrite(EVE_RST_PULSE,0U) is necessary for a start without power cycle, this line is in EVE_init() but commented out.

**Reply by RudolphRiedel** [reply]
And half an hour in I already "hate" Eclipse again... :-)

**Reply by suryazoox** [reply]
Hi @RudolphRiedel : I got it compiled with no errors. Now, I would need to include your main.c code into mine. :) Let me try it out. 

Also, just for a sanity check on the hardware connections:
NXP EVAL board J2-11 SCK(PTB2) - >  NHD Display's CN1 Pin 3
NXP EVAL board J2-9 SIN(PTB3) - >  NHD Display's CN1 Pin 5
NXP EVAL board J2- 7 SOUT (PTB4) -> NHD Display's CN1 Pin 4
NXP EVAL board J2- 5 CS (PTB5) -> NHD Display's CN1 Pin 6
NXP EVAL board J2- 3 PD (PTD14) -> NHD Display's CN1 Pin 8
NXP EVAL board J3- 7 (3.3V) -> NHD Display's CN1 Pin 1
NXP EVAL board J3- 20 (GND) -> NHD Display's CN1 Pin 20

**Reply by RudolphRiedel** [reply]
I can not recommend to use the 3.3V from the eval board.
There is only a linear regulator on the S32K144EVB as far as I can tell, this is U15. And it can "only" supply 500mA - if it is not shutting down with less load on the board.
The NHD 7 probably draws up to 130mA for the logic alone plus up to 760mA for the backlight.

NXP EVAL board J2-11 SCK(PTB2) - > NHD Display's CN1 Pin 3 - SCK, ok
NXP EVAL board J2-9 SIN(PTB3) - > NHD Display's CN1 Pin 5 - MOSI, Input - not ok
NXP EVAL board J2- 7 SOUT (PTB4) -> NHD Display's CN1 Pin 4 - MISO, Output - not ok
NXP EVAL board J2- 5 CS (PTB5) -> NHD Display's CN1 Pin 6 - CS, ok
NXP EVAL board J2- 3 PD (PTD14) -> NHD Display's CN1 Pin 8 - PD, ok

**Reply by RudolphRiedel** [reply]
I copied my old project in the workspace to a new project and added the most recent files like this:
![grafik](https://user-images.githubusercontent.com/31180093/221028908-aa79bb7c-557f-4ba5-9b06-652e0b3e49d1.png)

I also used the newer versions of the tft_xx files from my examples folder.
In main.c I had to change the include for EVE.h to
#include "EmbeddedVideoEngine/EVE.h"

And at least it compiles like this using the most current library files.

**Reply by suryazoox** [reply]
**so in this the CN1 Pin 4 needs to be connected to J2-9 and CN1 Pin 5 to J2-11 ???**  Can you please confirm? 

On the Eval board: I am using a power adapter for the backlight. 
I am using the jumper on J10 on the eval board to make it 3.3 V and also removed the R64 0 ohm resistor as well. 

Let me try using the recent files then and give it a shot.

**Reply by RudolphRiedel** [reply]
> **so in this the CN1 Pin 4 needs to be connected to J2-9 and CN1 Pin 5 to J2-11 ???** Can you please confirm?

Yes, I can confirm, I was using PTB3 for MISO and PTB4 for MOSI.

**Reply by suryazoox** [reply]
Thank you. 

So i made the changes. And the s32 compiles with no errors. 

When I run it, I see red the led turned on the eval board but nothing powers up on the LCD screen. Have you faced similar issues?

**Reply by RudolphRiedel** [reply]
Just a quick note before I have to leave for work, I was greeted with a black screen many times now when trying to bring up new hardware. :-)
At this point I am connecting my logic analyzer and check that the SPI is actually doing what it is supposed to do - do you have a logic analyzer?
I can have a look what my board is doing later this afternoon but I am not quite sure how to share the project properly when I get it to work, the last time I checked it the project-folder was using absolute paths all over and could not be moved.

**Reply by suryazoox** [reply]
Hi @RudolphRiedel : Yes, I am trying using a sale logic analyzer to see how the communication works. Right now, I am still debugging. 
Will keep you posted soon.

**Reply by RudolphRiedel** [reply]
I just connected the RVT50UQ I was originally using with the S32K144EVal board without changing the software.
NXP EVAL board J2-11 SCK(PTB2) - > SCK
NXP EVAL board J2-9 SIN(PTB3) - > MISO
NXP EVAL board J2- 7 SOUT (PTB4) -> MOSI
NXP EVAL board J2- 5 CS (PTB5) -> CS
NXP EVAL board J2- 3 PD (PTD14) -> PD
NXP EVAL board J6- 12 (GND) -> GND
And 7V from my PSU.

I used one of these: https://github.com/RudolphRiedel/EVE_display-adapter/tree/master/L-D5019-08

>NXP EVAL board J3- 20 (GND) -> J3 only has 16 Pin and GND would on 11 and 13.

This probably only was a typo?

And it works.
Now I switch projects and displays and flash the board with the new software.

Edit: so far I could not flash the board, some unclear issue with GDB.

**Reply by suryazoox** [reply]
Very Nice. Thank you. Can you please confirm the following;
1. Did you also set Ports in the components library like this? 
![image](https://user-images.githubusercontent.com/111003941/221264105-79dc784f-b43f-4d97-9832-5506aa9fcfd5.png)
![image](https://user-images.githubusercontent.com/111003941/221264317-d065946e-054f-41cc-ac7a-19f5d985d1e7.png)

2. Yes, It was a typo. 
3. Can you share your project with me? I know I can use it directly but I can create a new project and cut and copy the files to make it work. 
4. Also, did you power up the Back light pin as well ?

**Reply by RudolphRiedel** [reply]
1. I did not use the component library, I put my own code into EVE_init_spi in EVE_target.c. Can't put my finger on it but most likely because the component library was more bloat than functionality.
3.  [EVE_Test_S32K144_new.zip](https://github.com/RudolphRiedel/FT800-FT813/files/10827993/EVE_Test_S32K144_new.zip)
4. My adapter board powers logic and backlight with 3.3V using a switching regulator
5. Still no luck with flashing the software, annother bad experience with S32 DS in particular and Eclipse :-(

**Reply by suryazoox** [reply]
So you are not able to power up the LCD display screen? Did i understand that correct?

**Reply by RudolphRiedel** [reply]
It worked all right with the software that still was on the board.
Now I made it flash the new software from the new project and it is up and running from the current files.
The display is selected by setting a define. Properties - C/C++ Build - Settings - Standard S32DS C Compiler - Preprocessor.
![grafik](https://user-images.githubusercontent.com/31180093/221273602-5fdbf44f-b5da-4214-bec4-baed525f8fd7.png)

![S32K_EVE3_50G](https://user-images.githubusercontent.com/31180093/221273426-13909f7c-3645-4e27-800f-8fe3e3a71b01.jpg)

**Reply by suryazoox** [reply]
The code compiles and the EVB S32144K blinks the red LED ON/OFF however, I am still unable to view anything on the display. 

Not sure why this happening.

**Reply by RudolphRiedel** [reply]
Please attach a trace from the Salea of the first two seconds after reset / power up. Preferably with PD included.

**Reply by suryazoox** [reply]
https://drive.google.com/drive/folders/1V7fuoHU-gAGsDELYvUGxgM8qd5O55GQZ?usp=share_link


Here is the link for the trace.

**Reply by RudolphRiedel** [reply]
You can just .zip files to put them up here:
[RVT101H_startup.zip](https://github.com/RudolphRiedel/FT800-FT813/files/10854464/RVT101H_startup.zip)
:-)

**Reply by suryazoox** [reply]
Hi Rudolph, 
were you able to view the trace files?

**Reply by RudolphRiedel** [reply]
Nope, the folder is access restricted.

**Reply by suryazoox** [reply]
now you should have acess to the link.

**Reply by RudolphRiedel** [reply]
Nope, still the same, I can only ask for permission to access but I already did that.
Why not paste just the .sal file packed in a .zip here?

**Reply by RudolphRiedel** [reply]
Here is another one to take a peek: 
[RP2040_EVE3-50G.zip](https://github.com/RudolphRiedel/FT800-FT813/files/10854657/RP2040_EVE3-50G.zip)

Btw., the .sal files are actually renamed .zip files so renaming these to attach them would work as well.
